packages/svelte-ux/src/lib/components/CopyButton.svelte

```diff
@@ -25,11 +29,15 @@
   {...restProps}
   class={cls('CopyButton', settingsClasses.root, className)}
   on:click={() => {
-    navigator.clipboard.writeText(value);
-    showMessage = true;
+    let copyValue = typeof value === 'function' ? value() : value;
+    navigator.clipboard.writeText(copyValue);
+    dispatch('click');
```

Comment:
Instead of using `dispatch` maybe just add another `on:click` to forward the event.  It's a pattern I've used elsewhere ([NavItem](https://github.com/techniq/svelte-ux/blob/main/packages/svelte-ux/src/lib/components/NavItem.svelte#L42), [ToggleButton](https://github.com/techniq/svelte-ux/blob/main/packages/svelte-ux/src/lib/components/ToggleButton.svelte#L21))

Reply:
Yeah I go back and forth on that. Ended up doing it this way to be more explicit about the event firing after the clipboard has been copied, instead of relying on the undocumented behavior that the event handlers fire in the order they are listed in the template. (Not that it's likely to change at this point)

Also this is very slightly closer to how it will be in Svelte 5. :) Anyway I can change it if you still would like me to.

Reply:
Nope, makes sense to me to have the explicit ordering, especially to align with how things will be going forward with Svelte 5, which I believe becomes...

```svelte
<script>
  const { onclick } = $props;
</script>

<Button
  onclick={() => {
    // ...
    onclick();
  }}
/>
```

(or something like that.  just wrote that in a github comment without checking too much).